zone/client.cpp

```diff
@@ -10129,3 +10137,437 @@ void Client::SetAFK(uint8 afk_flag) {
 	entity_list.QueueClients(this, outapp);
 	safe_delete(outapp);
 }
+
+void Client::SetShadowStepExemption(bool v)
+{
+	if (v == true)
+	{
+		uint32 cur_time = Timer::GetCurrentTime();
+		if ((cur_time - m_TimeSinceLastPositionCheck) > 1000)
+		{
+			float speed = (m_DistanceSinceLastPositionCheck * 100) / (float)(cur_time - m_TimeSinceLastPositionCheck);
+			int runs = GetRunspeed();
+			if (speed > (runs * RuleR(Zone, MQWarpDetectionDistanceFactor)))
+			{
+				printf("%s %i moving too fast! moved: %.2f in %ims, speed %.2f\n", __FILE__, __LINE__,
```

Comment:
I'd like to see this converted to using `fmt` https://fmt.dev/latest/index.html which we have built into all of our logging already

Please

* Create a new category in `eqemu_logsys.h`
* Create a new enum in `LogCategory` called `Cheat`
* Create a new `LogCategoryName` entry to match called `Cheat` (At the end of the list)

Under `eqemu_logsys_log_aliases.h` we'll need new alias macros (Macros tremendously help avoid popping unnecessary data on the stack if we're not going to log anything, there was a huge performance penalty when we just used functions)

```cpp
#define LogCheat(message, ...) do {\
    if (LogSys.log_settings[Logs::Cheat].is_category_enabled == 1)\
        OutF(LogSys, Logs::General, Logs::Cheat, __FILE__, __func__, __LINE__, message, ##__VA_ARGS__);\
} while (0)

#define LogCheatDetail(message, ...) do {\
    if (LogSys.log_settings[Logs::Cheat].is_category_enabled == 1)\
        OutF(LogSys, Logs::Detail, Logs::Cheat, __FILE__, __func__, __LINE__, message, ##__VA_ARGS__);\
} while (0)
```
This also has built in ` __FILE__, __func__, __LINE__` printing if you enable it via rule `RuleB(Logging, PrintFileFunctionAndLine)`

This

```
printf("%s %i moving too fast! moved: %.2f in %ims, speed %.2f\n", __FILE__, __LINE__, m_DistanceSinceLastPositionCheck, (cur_time - m_TimeSinceLastPositionCheck), speed);
```

Becomes

```cpp
LogCheat(
	"[SetShadowStepExemption] Client [{}] moving too fast! Moved [{}] in [{}](s) speed [{}]",
	GetCleanName(),
	m_DistanceSinceLastPositionCheck, 
	(cur_time - m_TimeSinceLastPositionCheck), 
	speed
);
```
Similarly, all other prints in this code need to be converted to use this logging category so it can be turned on / off as needed. Don't worry about having an excess of logs. If the log is extra noisy just use the `LogCheatDetail` alias (level 3)